magma/syntax/sequential.py

```diff
@@ -348,15 +312,33 @@ def _sequential(
     magma_name = gen_free_name(call_def, defn_env, 'm')
     defn_env[magma_name] = m
 
-    inputs, output_type = get_io(call_def)
-    io_args = gen_io_args(inputs, output_type, async_reset, magma_name)
+    inputs = []
+    outputs = []
+    tuple_out = False
+    for k, v in cls.__call__.__annotations__.items():
+        if k == 'return':
+            if isinstance(v, tuple):
+                tuple_out = True
+                for idx, t in enumerate(v):
+                    t_name = gen_free_name(call_def, defn_env, 'T')
+                    outputs.append((f'O{idx}', t_name))
+                    defn_env[t_name] = t
+            else:
+                t_name = gen_free_name(call_def, defn_env, 'T')
+                outputs.append(('O', t_name))
+                defn_env[t_name] = v
+        else:
+            t_name = gen_free_name(call_def, defn_env, 'T')
+            defn_env[t_name] = v
+            inputs.append((k, t_name))
```

Comment:
Can we move this into a helper function just for organization?

Reply:
Will do